oml/datasets/audios.py

```diff
@@ -0,0 +1,392 @@
+        return audio
+
+    def __getitem__(self, item: int) -> Dict[str, Union[FloatTensor, int]]:
+        audio_tensor = self.get_audio(item)
+        data = {
+            self.input_tensors_key: audio_tensor,
+            self.index_key: item,
+        }
+        for key, record in self.extra_data.items():
+            if key in data:
+                raise ValueError(f"<extra_data> and dataset share the same key: {key}")
+            else:
+                data[key] = record[item]
+
+        return data
+
+    def __len__(self) -> int:
+        return len(self._paths)
+
+    def visualize(self, item: int, color: TColor = BLACK) -> np.ndarray:
+        """
+        Visualize an audio file.
+
+        Args:
+            item: Dataset item index.
+            color: Color of the plot.
+
+        Returns:
+            Array representing the image of the plot.
+        """
+        audio = self.get_audio(item)
+        spec_repr = self._spectral_function(audio)
+        return visualize_audio(spec_repr=spec_repr, color=color)
+
+    def visualize_with_player(self, item: int, color: TColor = BLACK) -> str:
+        """
+        Visualize an audio file in HTML markup.
+
+        Args:
+            item: Dataset item index.
+            color: Color of the plot.
+
+        Returns:
+            HTML markup with spectral representation image and audio player.
+        """
+        audio = self.get_audio(item)
+        spec_repr = self._spectral_function(audio)
+        return visualize_audio_with_player(audio=audio, spec_repr=spec_repr, sample_rate=self._sample_rate, color=color)
+
+
+class AudioLabeledDataset(DFLabeledDataset, IVisualizableDataset):
+    """
+    The dataset of audios having their ground truth labels.
+
+    """
+
+    _dataset: AudioBaseDataset
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        dataset_root: Optional[Union[str, Path]] = None,
+        extra_data: Optional[Dict[str, Any]] = None,
+        input_tensors_key: str = INPUT_TENSORS_KEY,
+        index_key: str = INDEX_KEY,
+        labels_key: str = LABELS_KEY,
+        sample_rate: int = DEFAULT_SAMPLE_RATE,
+        max_num_seconds: float = MAX_AUDIO_LEN,
+        is_mono: bool = DEFAULT_IS_MONO,
+        spec_repr_func: Callable[[FloatTensor], FloatTensor] = default_spec_repr_func,
+    ):
+        """
+
+        Args:
+            df: DataFrame with input data.
+            dataset_root: Base path for audio files.
+            extra_data: Extra data to include in dataset items.
+            input_tensors_key: Key under which audio tensors are stored.
+            index_key: Key for indexing dataset items.
+            labels_key: Key under which labels are stored.
+            sample_rate: Sampling rate of audio files.
+            max_num_seconds: Duration to use from each audio file.
+            is_mono: If True, the audio will be downmixed to one channel; otherwise, it will remain unchanged.
```

Comment:
add symbols `` around True for readthedocs: if `True` ...

Reply:
done